conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosIndexer.scala

```diff
@@ -28,166 +27,42 @@ import tech.cryptonomic.conseil.indexer.config.{
 }
 import tech.cryptonomic.conseil.indexer.logging.LorreProgressLogging
 import tech.cryptonomic.conseil.indexer.tezos.TezosErrors._
-import tech.cryptonomic.conseil.indexer.tezos.TezosNodeOperator.LazyPages
 
 import scala.annotation.tailrec
-import scala.collection.SortedSet
 import scala.concurrent.duration.{Duration, _}
 import scala.concurrent.{Await, ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
+import tech.cryptonomic.conseil.indexer.tezos.processing.BakingAndEndorsingRightsProcessing
+import tech.cryptonomic.conseil.indexer.tezos.processing.AccountsEventsProcessing
+import tech.cryptonomic.conseil.indexer.tezos.processing.BakersProcessing
+import tech.cryptonomic.conseil.indexer.tezos.processing.AccountsProcessing
+import tech.cryptonomic.conseil.indexer.tezos.processing.BlocksProcessing
 
 /** * Class responsible for indexing data for Tezos BlockChain */
-class TezosIndexer(
+class TezosIndexer private (
+    ignoreProcessFailures: Boolean,
     lorreConf: LorreConfiguration,
-    tezosConf: TezosConfiguration,
-    callsConf: NetworkCallsConfiguration,
-    streamingClientConf: HttpStreamingConfiguration,
-    batchingConf: BatchFetchConfiguration
+    nodeOperator: TezosNodeOperator,
+    blocksProcessing: BlocksProcessing,
+    accountsProcessing: AccountsProcessing,
+    bakersProcessing: BakersProcessing,
+    rightsProcessing: BakingAndEndorsingRightsProcessing,
+    accountsEventsProcessing: AccountsEventsProcessing,
+    terminationSequence: () => Future[ShutdownComplete]
```

Comment:
Why is this a member when it can simply live a function?

Reply:
It's an input parameter whose actual call argument will need to close over local variables (essentially the resources to be freed by the call)
Those variables can't be closed over by a separate function.